### PR TITLE
ARROW-3088: [Rust] Use internal `Result<T>` type instead of `Result<T, ArrowError>`

### PR DESCRIPTION
**Pull request by paddyhoran**
Updates internal code to use existing arrow `Result` type.  Also, includes the following minor updates that do not deserve their own PR:
 - Removed `println!` statements that had been commented out but left in the code
 - Updated to use short-hand syntax where using identifiers that match the argument names when creating objects (i.e. `{field}` vs `{field: field}`)
 - Updated some `assert!` macros to more specific assert versions (`assert_eq!` and `assert_ne!`) to improve debug output

### CODE REVIEW

**Comment by paddyhoran** [comment]
cc @andygrove

**Review by sunchao — APPROVED** [review]
LGTM except the lint issue. But I'll wait until @andygrove to make the final decision :)

**Comment by andygrove** [comment]
LGTM too. I will now attempt to merge this.

**Review by andygrove — APPROVED** [review]

**Comment by andygrove** [comment]
The merge script isn't working for me. I've sent an email to the mailing list.

**Pull request closed without merging**
